### PR TITLE
(draft) feat: add cody usage survey on new dotcom signup

### PR DESCRIPTION
**Pull request by erzhtor**
This PR adds Cody usage survey on the new dotcom signup.

**NOTE** that It will not show new survey for existing users

## Test plan
- `sg start dotcom`
- Go to https://sourcegraph.test:3443/signup page and signup with a new account
- Check that you see a new popup form
- Submit it, and make sure you don't see it again, even when re-signing in again

## Screenshot

https://user-images.githubusercontent.com/6717049/236505928-123a1601-c62e-430f-ad68-0a51e13973c0.mov

### CODE REVIEW

File: cmd/frontend/hubspot/hubspotutil/hubspotutil.go [hunk 1]
```diff
@@ -23,6 +23,9 @@ var HubSpotAccessToken = env.Get("HUBSPOT_ACCESS_TOKEN", "", "HubSpot access tok
 // SurveyFormID is the ID for a satisfaction (NPS) survey.
 var SurveyFormID = "ee042306-491a-4b06-bd9c-1181774dfda0"
 
+// CodySurveyFormID is the ID for a Cody usage survey on dotcom users.
+var CodySurveyFormID = "todo: replace with a real ID"
```
Comment (erzhtor) [inline cmd/frontend/hubspot/hubspotutil/hubspotutil.go:27]: TODO: replace with a real Hubspot form ID once created. It should contain the following fields https://github.com/sourcegraph/sourcegraph/pull/51523/files#diff-b455047dfeea269a9fc594c07f6b64a433abc02860866c551f45a5161d8d6321R284-R288

File: cmd/frontend/graphqlbackend/user_usage_stats.go [hunk 2]
```diff
@@ -277,3 +280,43 @@ func exportPrometheusSearchRanking(payload json.RawMessage) error {
...
+	// Submit form to HubSpot
+	if err := hubspotutil.Client().SubmitForm(hubspotutil.CodySurveyFormID, &codySurveySubmissionForHubSpot{
+		Email:         email,
+		IsForWork:     args.IsForWork,
+		IsForPersonal: args.IsForPersonal,
+	}); err != nil {
+		// Log an error, but don't return one if the only failure was in submitting survey results to HubSpot.
+		log15.Error("Unable to submit cody survey results to Sourcegraph remote", "error", err)
+		// fixme: remove this log line after we've confirmed that the survey is working
+		fmt.Printf("Unable to submit cody survey results to Sourcegraph remote: Email=%v, isForWork=%v, isForPersonal=%v\n", email, args.IsForWork, args.IsForPersonal)
```
Comment (erzhtor) [inline cmd/frontend/graphqlbackend/user_usage_stats.go:318]: TODO: remove once replacing hubspot ID and testing that it works correctly.

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.08% (+2.32 kb) | -0.05% (-7.53 kb) | -0.08% (-9.84 kb) 🔽 | -0.13% (-1) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/erzhtor/add-cody-survey-form-on-new-dotcom-signup/compare-report.html;tab=live_object) for a full comparison between the commits 62a3bf8 and 2299224 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by rrhyne — APPROVED** [review]
LGTM

**Review by dadlerj — APPROVED** [review]

**Pull request merged**